internal/impl/pure/processor_cached.go

```diff
@@ -159,29 +174,39 @@ func (proc *cachedProcessor) Process(ctx context.Context, msg *service.Message)
 		return nil, err
 	}
 
+	shouldCache := true
 	var collapsedBatch service.MessageBatch
 	for _, b := range resultBatch {
+		skip, err := shouldSkip(b, proc.skipOn)
+		if err != nil {
```

Comment:
I'm going to change this so that it's a soft-error where we log and set `shouldCache = false`, as this seems similar to the actual cache write errors where we treat them as annoyances where we ought to carry on.